jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/actions/DeployServerlessApplicationAction.kt

```diff
@@ -15,26 +18,50 @@ class DeployServerlessApplicationAction : AnAction(
         message("serverless.application.deploy"),
         null,
         AwsIcons.Resources.LAMBDA_FUNCTION) {
-    private var templateYamlRegex = Regex("template\\.y[a]?ml", RegexOption.IGNORE_CASE)
+    private val templateYamlRegex = Regex("template\\.y[a]?ml", RegexOption.IGNORE_CASE)
 
-    override fun actionPerformed(e: AnActionEvent?) {
+    override fun actionPerformed(e: AnActionEvent) {
 
-        val project = e?.getRequiredData(PlatformDataKeys.PROJECT) ?: throw Exception("Unable to determine project")
+        val project = e.getRequiredData(PlatformDataKeys.PROJECT)
 
-        val virtualFiles = e.getData(PlatformDataKeys.VIRTUAL_FILE_ARRAY) ?: throw Exception("Could not detect template file")
-        val samTemplateFile = virtualFiles[0]
+        val samTemplateFile = getSamTemplateFile(e) ?: throw Exception("Could not detect template file")
```

Comment:
Would we ever expect this to be surfaced to a customer? If so we should put it in `message`